react/features/toolbox/components/native/Toolbox.js

```diff
@@ -9,157 +9,168 @@ import {
- */
-const secondaryToolbarButtonStyles = {
-    iconStyle: styles.secondaryToolbarButtonIcon,
-    style: styles.secondaryToolbarButton,
-    underlayColor: 'transparent'
-};
+const TOOLBAR_BUTTON_SIZE_FACTOR = 0.8;
 
 /**
  * The type of {@link Toolbox}'s React {@code Component} props.
  */
 type Props = {
 
     /**
-     * The indicator which determines whether the toolbox is enabled.
+     * The indicator which determines whether the toolbox is visible.
      */
-    _enabled: boolean,
+    _visible: boolean,
 
     /**
-     * Flag showing whether toolbar is visible.
+     * The redux {@code dispatch} function.
      */
-    _visible: boolean
+    dispatch: Function
+};
+
+/**
+ * The type of {@link Toolbox}'s React {@code Component} state.
+ */
+type State = {
+
+    /**
+     * The detected width for this component.
+     */
+    width: number
 };
 
 /**
  * Implements the conference toolbox on React Native.
  */
-class Toolbox extends Component<Props> {
+class Toolbox extends Component<Props, State> {
+    state = {
+        width: 0
+    };
+
     /**
-     * Implements React's {@link Component#render()}.
+     *  Initializes a new {@code Toolbox} instance.
      *
      * @inheritdoc
-     * @returns {ReactElement}
      */
-    render() {
-        if (!this.props._enabled) {
-            return null;
-        }
+    constructor(props: Props) {
+        super(props);
 
-        const toolboxStyle
-            = isNarrowAspectRatio(this)
-                ? styles.toolboxNarrow
-                : styles.toolboxWide;
-
-        return (
-            <Container
-                style = { toolboxStyle }
-                visible = { this.props._visible } >
-                { this._renderToolbars() }
-            </Container>
-        );
+        this._onLayout = this._onLayout.bind(this);
     }
 
+    _onLayout: (Object) => void;
+
     /**
-     * Renders the toolbar which contains the primary buttons such as hangup,
-     * audio and video mute.
+     * Handles the "on layout" View's event and stores the width as state.
      *
+     * @param {Object} event - The "on layout" event object/structure passed
+     * by react-native.
      * @private
-     * @returns {ReactElement}
+     * @returns {void}
      */
-    _renderPrimaryToolbar() {
-        return (
-            <View
-                key = 'primaryToolbar'
-                pointerEvents = 'box-none'
-                style = { styles.primaryToolbar }>
-                <AudioMuteButton
-                    styles = { primaryToolbarButtonStyles }
-                    toggledStyles = { primaryToolbarToggledButtonStyles } />
-                <HangupButton styles = { hangupButtonStyles } />
-                <VideoMuteButton
-                    styles = { primaryToolbarButtonStyles }
-                    toggledStyles = { primaryToolbarToggledButtonStyles } />
-            </View>
-        );
+    _onLayout({ nativeEvent: { layout: { width } } }) {
```

Comment:
@saghul, this comment is not a request for action on this PR:

1. react-native's `onLayout` is known to be sub-optimal because it needs at least 1 `render`. I'm not thinking of not going that way but I can actually see the toolbar buttons shifting their size not only in the Android Emulator and Simulator but also on my Samsung S8. I presume not rendering the buttons until after the size is actually calculated is still going to result in a visible tap delay.

2. The non-hangup buttons are actually uncomfortably small on my Samsung S8.

Reply:
Good points.

1. I don't think the delay would be noticeable, since it's just the first tap. We can give it a shot and see if it feels ok.

2. Can you share a screenshot? Did you see much margin available on the sides? We can try to squeeze things a bit to make the buttons larger maybe.

Reply:
> I don't think the delay would be noticeable, since it's just the first tap. We can give it a shot and see if it feels ok.

@saghul, yes please, let's try (a) not rendering the buttons until the size is calculated or (b) defaulting to a non-zero `width` (which is very likely much easier said than done but it could be as wild as passing a default width from a ascendant Component which gets its width earlier than the toolbox, the device's width). I can definitely see the buttons shifting their size on my iPhone 7 as well.

> Can you share a screenshot? Did you see much margin available on the sides? We can try to squeeze things a bit to make the buttons larger maybe.

I can work on a screenshot later on but I remember the following: I don't have any space on the left and right sides of the toolbar i.e. the outermost buttons feel like they're touching the ends of the display and with the Samsung S8's curved display I feel like a couple of pixels more and the outermost buttons will spill outside the display ;)  However, there's ample space between the buttons. So I'd be more comfortable with slightly bigger buttons with less space in between them (I presume).

Reply:
:+1:

Reply:
https://github.com/jitsi/jitsi-meet/pull/2993
https://github.com/jitsi/jitsi-meet/pull/2994